### PR TITLE
Add support for additional multiplatform targets

### LINKED ISSUE #77: Feature request: mingwX86 target
Would it be possible/easy to add target mingwX86?

PS: thanks for such an elegant design.

### PR DESCRIPTION
**Pull request by 05nelsonm**
This PR includes the following modifications:

- Adds support for additional multiplatform targets to modules `kotlin-result` and `kotlin-result-coroutines`:
     - `iosArm32`
     - `tvosArm64`
     - `tvosX64`
     - `tvosSimulatorArm64`
     - `watchosArm32`
     - `watchosArm64`
     - `watchosX64`
     - `watchosX86`
     - `watchosSimulatorArm64`
- Adds support for additional multiplatform targets to module `kotlin-result` (in addition to above):
     - `androidNativeArm32`
     - `androidNativeArm64`
     - `androidNativeX64`
     - `androidNativeX86`
     - `linuxArm32Hfp`
     - `linuxArm64`
     - `linuxMips32`
     - `linuxMipsel32`
     - `mingwX86`
     - `wasm32`
- Updates GitHub Actions test/build & publishing for all targets
- Updates the `README` with new supported platforms

Closes #77

### CODE REVIEW

**Comment by 05nelsonm** [comment]
Builds on all platforms

Screenshot from 2023-01-05 20-54-30

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -91,11 +91,11 @@ jobs:
...
-      - run: ./gradlew publishLinuxX64PublicationToMavenRepository
+      - run: ./gradlew $(./gradlew tasks | grep "PublicationToMavenRepository" | grep -e "publishAndroid" -e "publishJs" -e "publishJvm" -e "publishKotlinMultiplatform" -e "publishLinux" -e "publishWasm" | cut -d ' ' -f 1)
         if: matrix.os == 'ubuntu-latest'
 
-      - run: ./gradlew publishMingwX64PublicationToMavenRepository
+      - run: ./gradlew $(./gradlew tasks | grep "PublicationToMavenRepository" | grep "publishMingw" | cut -d ' ' -f 1)
         if: matrix.os == 'windows-latest'
 
-      - run: ./gradlew publish
+      - run: ./gradlew $(./gradlew tasks | grep "PublicationToMavenRepository" | grep -e "publishIos" -e "publishMacos" -e "publishTvos" -e "publishWatchos" | cut -d ' ' -f 1)
```
[the remark this answers is not in the document]
Comment (michaelbull) [reply .github/workflows/ci.yaml:100]: I am not a fan of the complexity going on with these shell commands. If this complexity is to live anywhere it should be in the Gradle scripts themselves, such that the commands are multi-platform and not specific to a given shell implementation (in this case, bash).

**Comment by michaelbull** [comment]
I think we will need to add a disclaimer to the coroutines section explaining the limitations on build targets in comparison with the core library, as the unfortunate platform compatibility differences will trip some people up.

**Comment by 05nelsonm** [comment]
> I think we will need to add a disclaimer to the coroutines section explaining the limitations on build targets in comparison with the core library, as the unfortunate platform compatibility differences will trip some people up.

Resolved in commit 7709128

I also fixed the badges in commit fff3239 which I fudged (sorry). Thoughts?

File: .github/workflows/ci.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (05nelsonm) [reply .github/workflows/ci.yaml]: Resolved in commit 09d3028

**Comment by michaelbull** [comment]
Looking a lot better, thanks! I'll approve and run this to see how it fares on GitHub actions. As a disclaimer I am unsure when I will be available to actually merge and deploy this to Maven Central, I am very busy at the moment - but please don't take that as me leaving the PR dormant. I also need to get around to updating all the dependencies at some point, which has been on my backlog for a while.

**Comment by 05nelsonm** [comment]
> Looking a lot better, thanks! I'll approve and run this to see how it fares on GitHub actions. As a disclaimer I am unsure when I will be available to actually merge and deploy this to Maven Central, I am very busy at the moment - but please don't take that as me leaving the PR dormant. I also need to get around to updating all the dependencies at some point, which has been on my backlog for a while.

Right on. Sounds good, thanks!

**Comment by 05nelsonm** [comment]
Heyo, wanted to just note that Kotlin `1.8.0` added a new target; `watchosDeviceArm64`. Since coroutines hasn't had a release in forever, would simply need to:
 - Add it to the `kotlin-result` module, linking it to the intermediate source set `watchosMain` (and `watchosTest` if necessary)
 - Update CI publishing task with `publishWatchosDeviceArm64PublicationToMavenRepository`

**Comment by michaelbull** [comment]
@05nelsonm I've squashed and rebased the changes onto master in https://github.com/michaelbull/kotlin-result/commit/6f86d20d53adaa814ca68cbb91946abcf7be2c2a

Could you please try to depend on the latest SNAPSHOT build and see if the artifacts you're expecting for each platform are consumable?

**Comment by 05nelsonm** [comment]
> @05nelsonm I've squashed and rebased the changes onto master in [6f86d20](https://github.com/michaelbull/kotlin-result/commit/6f86d20d53adaa814ca68cbb91946abcf7be2c2a)
> 
> Could you please try to depend on the latest SNAPSHOT build and see if the artifacts you're expecting for each platform are consumable?

:+1: 

image

**Comment by michaelbull** [comment]
Thanks @05nelsonm. I'll look at the other PR during the week and also try to update dependencies, then see if I can make a release maybe on the weekend - but at least you should be able to depend upon the SNAPSHOT in the mean time.

**Comment by michaelbull** [comment]
Released in v1.1.17.

**Pull request closed without merging**